app/src/main/java/com/google/android/soundchecker/FilePlayerService.kt

```diff
@@ -19,31 +19,63 @@ package com.google.android.soundchecker
 import android.app.Notification
 import android.app.NotificationChannel
 import android.app.NotificationManager
+import android.app.PendingIntent
 import android.app.Service
 import android.content.Intent
 import android.content.pm.ServiceInfo
 import android.os.IBinder
+import android.util.Log
 import androidx.core.app.NotificationCompat
 
 class FilePlayerService : Service() {
     companion object {
+        private const val TAG = "FilePlayerService"
+
         private const val NOTIFICATION_CHANNEL_ID = "SoundChecker"
         private const val CHANNEL_NAME = "FilePlayer"
         private const val NOTIFICATION_ID = 1
 
         const val TITLE = "title"
         const val SONG_NAME = "song_name"
+
+        const val ACTION_START_SERVICE = "soundchecker.action.START_FILE_PLAYER_SERVICE"
+        const val ACTION_STOP_SERVICE = "soundchecker.action.STOP_FILE_PLAYER_SERVICE"
+        const val ACTION_START_PLAYBACK = "soundchecker.action.START_PLAYBACK"
+        const val ACTION_PLAYBACK_STARTED = "soundchecker.action.PLAYBACK_STARTED"
+        const val ACTION_STOP_PLAYBACK = "soundchecker.action.STOP_PLAYBACK"
+        const val ACTION_PLAYBACK_STOPPED = "soundchecker.action.PLAYBACK_STOPPED"
     }
 
     private var mTitle: String = ""
     private var mSongName: String = ""
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
-        startForeground(NOTIFICATION_ID, getNotification(),
-            ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK)
-        if (intent != null) {
-            mTitle = intent.getStringExtra(TITLE).toString()
-            mSongName = intent.getStringExtra(SONG_NAME).toString()
+        when (intent?.action) {
+            ACTION_START_SERVICE -> {
+                startForeground(
+                    NOTIFICATION_ID, getNotification(""),
+                    ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK
+                )
+            }
+            ACTION_PLAYBACK_STARTED -> {
+                mTitle = intent.getStringExtra(TITLE).toString()
+                mSongName = intent.getStringExtra(SONG_NAME).toString()
+                startForeground(NOTIFICATION_ID, getNotification(ACTION_STOP_PLAYBACK),
```

Comment:
Why is this one ACTION_STOP_PLAYBACK and the one below ACTION_START_PLAYBACK?

Reply:
When the playback is started, the action from the notification is to stop it. When the playback is stopped, the action will now change to start the playback.